### ISSUE TITLE
Swap out tailable cursor with mongo queue on alpine post

### ISSUE THREAD

**Issue by githubbob42** [githubbob42/mingle2github2#2753]
**[Mingle Card: 3037](https://liquidframeworks.mingle.thoughtworks.com/projects/alpine_mobile/cards/3037)**
Narrative
---------

We want to support multiple alpine post processes running concurrently to reduce the backlog of upload packets waiting to be processed. Currently alpine post uses a tailable cursor, and we would need to swap it out with a mongo queue so that both alpine post processes don't attempt to process the same item.

Acceptance Criteria
-------------------

_Enter the Acceptance Criteria here._

Analysis
--------

After talking with Craig, supporting multiple post processes is a bit more involved than supporting multi-process for fetch.  With post we need to ensure that multiple post process grab the same packet to process.  We also need to ensure that multiple packages from the same user, get run sequentially.   

Let's implement the multi-process for post in another card and implement a proper queue first.  This should fix cards #3231 and #3284.

Concurrency/Sequentiality Issues to consider:

*   we will need to run both the old post and new post processes at the same time for some set time in order to clean out the old/existing packets and to run any packets coming in on the new queue.
    *   because of this we'll probably need to create a new collection name for the new messages queue.
*   not sure at this time how we're going to do this... i.e. new and old post process at same time.
*   What about existing sync packets (failed and unprocessed)?
    *   Can we write a tool to move failed pkts to the deadQueue?
    *   Can we move unprocessed messages to new queue?  Is this even necessary?
    *   Will moving old messages to new queue(s) eliminate the need for old/new concurrency?

**TASKS**:

alpine-mobile

*   update mobile to write to new queue/collection (separate from "messages" collection)

alpine-post:

*   convert post to use proper queue (mongodb-queue)
*   set # of retries to 1 (run only once because if a packet fails halfway through, it would try to rerun the entire packet again - rerunning events it already processed)
*   after successfully processing a packet, add an entry to a new queue/collection for alpine-analytics

sync error processing to use new deadQueue (created by mongodb-queue)

*   remove indexes from messages selection
*   add indexes to deadQueue collection (this is where failed sync packets go)

Add new processor that processes new deadQueue.  For each queue item:

*   add to "erroredPackets" collection (non-queue), with an appropriate projection (probably just strip off queueing properties, etc?).  This is because we should not read/write a queue outside of using the queue module.
*   maybe eventually move auditLogging to this process, to separate it from the actual post processing.  To be done under another card (with it's own analysis) if we do this.

sync error processing to use new "erroredPackets" collection (non-queue)

*   remove indexes from messages collection (is this even necessary?  we'll probably just drop the messages collection)
*   add indexes to "erroredPackets" collection
*   read from "erroredPackets" collection as normal(?)
*   if packet is rerun successfully (or maybe on each attempt) should we right the AuditLog?  Currently we just mark the original (errored) packet as processed=true (which isn't necessarily correct)

alpine-analytics

*   read new queue (created by alpine-post) of successfully processes packets.
*   perhaps the post processor should write the the analytics queue any events it finds until an error occurs processing the packet.

Do we need to do anything with cleanup-sync-packets process?

Related Stories
---------------

 #3231 and #3284

Tasks
-----

{{table query: SELECT Number, Name, Owner, 'Task Status' WHERE Type = Task and Story = THIS CARD}}

Defects
-------

{{ table query: SELECT Number, Name, Owner, 'Status' WHERE Type = Defect and 'Related Story' = THIS CARD }}

Test Plan
---------
